### PR TITLE
[Storage Blob Service] ORS

### PR DESCRIPTION
**Pull request by JinmingHu-MSFT**
closes https://github.com/Azure/azure-sdk-for-cpp/issues/302

### CODE REVIEW

File: sdk/storage/inc/blobs/protocol/blob_rest_client.hpp [hunk 1]
```diff
@@ -4187,6 +4251,7 @@ namespace Azure { namespace Storage { namespace Blobs {
           k_Sealed,
           k_xmsblobsequencenumber,
           k_Metadata,
+          k_OrMetadata,
```
Comment (katmsft) [inline sdk/storage/inc/blobs/protocol/blob_rest_client.hpp:4254]: Is `OR` an acronym?

File: sdk/storage/src/datalake/datalake_path_client.cpp [hunk 2]
```diff
@@ -294,14 +294,13 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
...
+    ret.AccessTierChangeTime = std::move(result->AccessTierChangedOn);
     ret.CopyId = std::move(result->CopyId);
     ret.CopySource = std::move(result->CopySource);
     ret.CopyStatus = std::move(result->CopyStatus);
     ret.CopyProgress = std::move(result->CopyProgress);
-    ret.CopyCompletionTime = std::move(result->CopyCompletionTime);
-    return Azure::Core::Response<GetPathPropertiesResult>(
-        std::move(ret), result.ExtractRawResponse());
+    ret.CopyCompletionTime = std::move(result->CopyCompletedOn);
+    return Azure::Core::Response<GetPathPropertiesResult>(std::move(ret), result.ExtractRawResponse());
```
Comment (katmsft) [inline sdk/storage/src/datalake/datalake_path_client.cpp:303]: Strange, my clang-formatter thinks this line is too long.

File: sdk/storage/inc/blobs/protocol/blob_rest_client.hpp (hunk 1 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/protocol/blob_rest_client.hpp:4254]: in xml it's called `OrMetadata`

File: sdk/storage/src/datalake/datalake_path_client.cpp (hunk 2 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/src/datalake/datalake_path_client.cpp:303]: i'm going to fix that

Comment (JinmingHu-MSFT) [reply sdk/storage/src/datalake/datalake_path_client.cpp]: fixed

**Comment by JinmingHu-MSFT** [comment]
> Why isn't there UT covering the change?

No UT, targeting tested

**Pull request merged**